### ISSUE TITLE
Stateless Components not working with SSR

### ISSUE THREAD

**Issue by rattrayalex** [apollostack/react-apollo#297]
This line: https://github.com/apollostack/react-apollo/blame/master/src/server.ts#L48

throws this error: `ComponentClass is not a constructor`
with this stack:

```
TypeError: ComponentClass is not a constructor at getQueriesFromTree (/Users/alex/djcode/another_example_app/frontend/src/server.ts:48:23) at getQueriesFromTree (/Users/alex/djcode/another_example_app/frontend/src/server.ts:67:5) at /Users/alex/djcode/another_example_app/frontend/src/server.ts:73:54 at forEachSingleChild (/Users/alex/djcode/another_example_app/frontend/node_modules/react/lib/ReactChildren.js:52:8) at traverseAllChildrenImpl (/Users/alex/djcode/another_example_app/frontend/node_modules/react/lib/traverseAllChildren.js:69:5) at traverseAllChildren (/Users/alex/djcode/another_example_app/frontend/node_modules/react/lib/traverseAllChildren.js:164:10) at Object.forEachChildren [as forEach] (/Users/alex/djcode/another_example_app/frontend/node_modules/react/lib/ReactChildren.js:72:3) at getQueriesFromTree (/Users/alex/djcode/another_example_app/frontend/src/server.ts:73:14) at getQueriesFromTree (/Users/alex/djcode/another_example_app/frontend/src/server.ts:67:5) at getQueriesFromTree (/Users/alex/djcode/another_example_app/frontend/src/server.ts:67:5)
```

This occurs only with Stateless Components, which work fine when rendered on the client but break when rendered on the server with react-apollo and react-router. 

This is the immediately surrounding code:

``` js
  const App = (
    <ApolloProvider client={client} >
      <RouterContext {...renderProps} />
    </ApolloProvider>
  )

  const { markup, initialState } = await renderToStringWithData(App)
```

When I only have class-based components, everything works. When I have a Stateless component anywhere in my hierarchy, I get the above error. 
### Version
- apollo-client@0.5.0
- react-apollo@0.5.14
- react@15.3.2

**Comment by rattrayalex** [comment]
This also occurs when `<RouterContext />` is not used

**Comment by tmeasday** [comment]
That's interesting, I'm pretty sure it does work in some cases (for instance GitHunt has some SFCs). In any case, I'm going to refactor this code soon, thanks for reporting it.

**Comment by rattrayalex** [comment]
Awesome, thanks @tmeasday ! Looking forward.

**Comment by tmeasday** [comment]
Fixed in 0.6.0 (I think?)
